scipy/stats/_continuous_distns.py

```diff
@@ -1640,8 +1634,220 @@ def _munp(self, n, c):
 
     def _entropy(self, c):
         return -_EULER / c - np.log(c) + _EULER + 1
+
+weibull_max = weibull_max_gen(b=0.0, name='weibull_max')
+
+# Public methods to be deprecated in frechet_r and frechet_l:
+# ['__call__', 'cdf', 'entropy', 'expect', 'fit', 'fit_loc_scale', 'freeze',
+#  'interval', 'isf', 'logcdf', 'logpdf', 'logsf', 'mean', 'median', 'moment',
+#  'nnlf', 'pdf', 'ppf', 'rvs', 'sf', 'stats', 'std', 'var']
+
+_frechet_r_deprec_msg = """\
+The distribution `frechet_r` is a synonym for `weibull_min`; this historical
+usage is deprecated because of possible confusion with the (quite different)
+Frechet distribution.  To preserve the existing behavior of the program, use
+`scipy.stats.weibull_min`.  For the Frechet distribution (i.e. the Type II
+extreme value distribution), use `scipy.stats.frechet`."""
```

Comment:
Now that you've split the PR, this needs a tweak. ``stats.frechet`` doesn't exist.

Reply:
Same comment applies to the deprecation message for ``frechet_l``

Reply:
Oops, missed that one.  I just pushed a fix.